app/scripts/controllers/edit/buildConfig.js

```diff
@@ -522,6 +549,23 @@ angular.module('openshiftConsole')
       return sourceMap;
     };
 
+    $scope.addWebhookTrigger = function(webhookLabel) {
+      if (!webhookLabel) {
+        return;
+      }
+      var webhook = {
+        disabled: false,
+        data: {
+          type: webhookLabel
```

Comment:
Ah, I see that this is the API type value. Sorry didn't realize. I suggest:

```js
      options: [{
        type: 'GitHub',
        propterty: 'github'
      }, { ... }]
```

And then this function should take in parameter `type`.